File: zhitan-risk-agent/backend/schemas/risk.py
from __future__ import annotations

from typing import Any, Literal, Optional
from pydantic import BaseModel, Field

QueryType = Literal["company_name", "job_link", "screenshot"]
DimensionName = Literal["scale", "capital", "legal", "online", "position"]
Confidence = Literal["high", "medium", "low"]
SourceType = Literal["api", "search", "llm_infer", "mock"]
RiskLevel = Literal["low", "medium", "high", "insufficient"]
DataSufficiency = Literal["sufficient", "partial", "insufficient"]
Severity = Literal["info", "warning", "critical"]
QuestionType = Literal["verify", "probe", "decision"]
QualityGrade = Literal["excellent", "good", "poor"]
FeedbackTrackType = Literal["current_employee", "former_employee", "interviewee"]


class PositionInfo(BaseModel):
    title: Optional[str] = None
    salary: Optional[str] = None
    stage: Optional[str] = None
    scale: Optional[str] = None


class QueryContext(BaseModel):
    queryId: str
    queryType: QueryType
    companyName: str
    standardName: Optional[str] = None
    creditCode: Optional[str] = None
    aliases: list[str] = Field(default_factory=list)
    positionInfo: Optional[PositionInfo] = None
    createdAt: str


class RiskSignal(BaseModel):
    type: str
    title: str
    description: str
    source: str
    sourceType: SourceType
    confidence: Confidence
    fetchedAt: str
    rawValue: Optional[str] = None
    baseScore: float = 0
    explanation: Optional[str] = None
    suggestion: Optional[str] = None


class RiskDimension(BaseModel):
    name: DimensionName
    riskScore: float
    confidence: Confidence
    signals: list[RiskSignal] = Field(default_factory=list)
    summary: str


class TopSignal(BaseModel):
    signal: str
    severity: Severity
    explanation: str
    source: str
    confidence: Confidence
    suggestion: Optional[str] = None


class RiskConflict(BaseModel):
    dimensions: list[str]
    description: str


class RiskVerdict(BaseModel):
    overallScore: float
    riskLevel: RiskLevel
    dataSufficiency: DataSufficiency
    topSignals: list[TopSignal] = Field(default_factory=list)
    conflicts: list[RiskConflict] = Field(default_factory=list)
    summary: str


class InterviewQuestion(BaseModel):
    id: str
    type: QuestionType
    question: str
    relatedSignal: str


class FeedbackTrack(BaseModel):
    type: FeedbackTrackType
    title: str
    tags: list[str]
    summary: str
    count: int


class Quality(BaseModel):
    grade: QualityGrade
    warnings: list[str] = Field(default_factory=list)


class RiskCard(BaseModel):
    title: str
    theme: str
    levelText: str
    scoreText: str
    sufficiencyText: str
    summary: str


class RadarItem(BaseModel):
    label: str
    value: float


class RenderData(BaseModel):
    riskCard: RiskCard
    radar: list[RadarItem]
    loadingSteps: list[str]


class RiskReport(BaseModel):
    queryContext: QueryContext
    verdict: RiskVerdict
    dimensions: list[RiskDimension]
    interviewQuestions: list[InterviewQuestion]
    feedback: list[FeedbackTrack] = Field(default_factory=list)
    fallbackChecklist: list[str] = Field(default_factory=list)
    quality: Quality
    renderData: Optional[RenderData] = None
    disclaimer: str
    generatedAt: str


class QueryRequest(BaseModel):
    query: str
    queryType: Optional[QueryType] = None
    positionInfo: Optional[PositionInfo] = None


class CompanyShortcut(BaseModel):
    name: str
    type: str
    description: str


SampleCompany = dict[str, Any]

File: zhitan-risk-agent/frontend/src/types/risk.ts
export type QueryType = 'company_name' | 'job_link' | 'screenshot';
export type DimensionName = 'scale' | 'capital' | 'legal' | 'online' | 'position';
export type Confidence = 'high' | 'medium' | 'low';
export type RiskLevel = 'low' | 'medium' | 'high' | 'insufficient';
export type QuestionType = 'verify' | 'probe' | 'decision';

export interface PositionInfo {
  title?: string | null;
  salary?: string | null;
  stage?: string | null;
  scale?: string | null;
}

export interface QueryContext {
  queryId: string;
  queryType: QueryType;
  companyName: string;
  standardName?: string | null;
  creditCode?: string | null;
  aliases: string[];
  positionInfo?: PositionInfo | null;
  createdAt: string;
}

export interface RiskSignal {
  type: string;
  title: string;
  description: string;
  source: string;
  sourceType: 'api' | 'search' | 'llm_infer' | 'mock';
  confidence: Confidence;
  fetchedAt: string;
  rawValue?: string | null;
  baseScore: number;
  explanation?: string | null;
  suggestion?: string | null;
}

export interface RiskDimension {
  name: DimensionName;
  riskScore: number;
  confidence: Confidence;
  signals: RiskSignal[];
  summary: string;
}

export interface TopSignal {
  signal: string;
  severity: 'info' | 'warning' | 'critical';
  explanation: string;
  source: string;
  confidence: Confidence;
  suggestion?: string | null;
}

export interface RiskVerdict {
  overallScore: number;
  riskLevel: RiskLevel;
  dataSufficiency: 'sufficient' | 'partial' | 'insufficient';
  topSignals: TopSignal[];
  conflicts: Array<{ dimensions: string[]; description: string }>;
  summary: string;
}

export interface InterviewQuestion {
  id: string;
  type: QuestionType;
  question: string;
  relatedSignal: string;
}

export interface FeedbackTrack {
  type: 'current_employee' | 'former_employee' | 'interviewee';
  title: string;
  tags: string[];
  summary: string;
  count: number;
}

export interface RenderData {
  riskCard: {
    title: string;
    theme: string;
    levelText: string;
    scoreText: string;
    sufficiencyText: string;
    summary: string;
  };
  radar: Array<{ label: string; value: number }>;
  loadingSteps: string[];
}

export interface RiskReport {
  queryContext: QueryContext;
  verdict: RiskVerdict;
  dimensions: RiskDimension[];
  interviewQuestions: InterviewQuestion[];
  feedback: FeedbackTrack[];
  fallbackChecklist: string[];
  quality: { grade: 'excellent' | 'good' | 'poor'; warnings: string[] };
  renderData: RenderData;
  disclaimer: string;
  generatedAt: string;
}

export interface CompanyShortcut {
  name: string;
  type: string;
  description: string;
}
